### PR TITLE
support print Version

### LINKED ISSUE #650: edgecontroller and edge_core need support show current detail version
**What would you like to be added**:
edgecontroller and edge_core and other kubeedge binary need support print current detail version, such as 
```
Version: 2e77d11
Revision: 2e77d11631c0067e3366242a7fc9a6f5700bebc6
Goversion: go1.11
Branch: version
```
**Why is this needed**:

we can use this detail version info to debug or track some question.

and our bug report will more detailed。
such as
```

**Environment**:
- KubeEdge version:
- kubeedge Revision:
- GoVersion:
- Branch: 
- Hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Others:


```

### PR DESCRIPTION
**Pull request by kadisi**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

print binary version : git commit , version ,reversion, branch ,go version

fix #650 
**Special notes for your reviewer**:

### CODE REVIEW

**Comment by sids-b** [comment]
@kadisi can you please fix the CI ? Also please limit the changes only to support the printing of version in this PR for eg: (lets not change travis job names) in this PR.

**Comment by kadisi** [comment]
@sids-b PTAL

**Comment by rohitsardesai83** [comment]
@kadisi , can we have an issue to track this PR ?

**Comment by kadisi** [comment]
@rohitsardesai83   #650

File: keadm/Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # make kubeedge
 .PHONY: default kubeedge
 kubeedge:
-	go build -o kubeedge keadm.go
+	 @go build ${GO_GCFLAGS} ${GO_BUILD_FLAGS} -o $@ ${GO_LDFLAGS} ${GO_TAGS}  ./$<
```
Comment (subpathdev) [inline keadm/Makefile]: please remove $< you don't set an input file

File: edge/Makefile [hunk 2]
```diff
@@ -1,8 +1,9 @@
 
 # make edge_core
 .PHONY: default edge_core
+
 edge_core:
-	go build cmd/edge_core.go
+	@go build ${GO_GCFLAGS} ${GO_BUILD_FLAGS} ${GO_LDFLAGS} ${GO_TAGS}  cmd/edge_core.go $<
```
Comment (subpathdev) [inline edge/Makefile]: please remove $< you don't set an input file

File: keadm/Makefile (hunk 1 above)
Comment (kadisi) [reply keadm/Makefile]: ok

File: Makefile [hunk 3]
```diff
@@ -1,5 +1,33 @@
...
+
+PKG=github.com/kubeedge/kubeedge
+
+# Used to populate variables in version package.
+VERSION=$(shell git describe --match 'v[0-9]*' --dirty='.m' --always)
+REVISION=$(shell git rev-parse HEAD)$(shell if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi)
+GOVERSION=$(shell go version |awk -F ' ' '{printf $$3}')
+CURRENT_BRANCH=$(shell git symbolic-ref --short -q HEAD)
+
+BUILDTAGS ?=
```
Comment (subpathdev) [inline Makefile]: you can remove it, you do not set a value

**Comment by rohitsardesai83** [comment]
> @rohitsardesai83 #650

thanks @kadisi

**Comment by rohitsardesai83** [comment]
@kadisi , can you please output how the version shows up in the logs or on the command line after this change ?

**Comment by kadisi** [comment]
@rohitsardesai83 , yes ,  i know what you mean.   maybe this change is not perfect. the best is like kedam: 
this is it`s output:
```
./keadm/kubeedge version
Package: github.com/kubeedge/kubeedge
Version: ca145a1
Revision: ca145a1e9ad437e7b72e7194ce97524c2dd467e9
Goversion: go1.10.2
Branch: version

```
but about edgecontroller and edge_core , now it only output it`s logs:

this is edgecontroller output:
```

./cloud/edgecontroller
2019/06/13 04:14:20 WARN: empty configurtion from [CommandlineSource]
2019/06/13 04:14:20 WARN: empty configurtion from [MemorySource]
New file source added for configuration: /go/src/github.com/kubeedge/kubeedge/cloud/conf/controller.yamlNew file source added for configuration: /go/src/github.com/kubeedge/kubeedge/cloud/conf/logging.yamlNew file source added for configuration: /go/src/github.com/kubeedge/kubeedge/cloud/conf/modules.yaml/go/src/github.com/kubeedge/kubeedge/cloud/conf/logging.yamllogConfig:&{INFO edgecontroller.log [file stdout] true false  }
2019/06/13 04:14:20 RollingPolicy is empty, use default policy[size]
2019-06-13 04:14:20.288 +00:00 INFO core/module.go:41 module cloudhub registered
2019-06-13 04:14:20.347 +00:00 INFO config/buffer.go:36 Update controller.buffer.update-pod-status: 1024
2019-06-13 04:14:20.349 +00:00 INFO config/buffer.go:43 Update controller.buffer.update-node-status: 1024
2019-06-13 04:14:20.352 +00:00 INFO config/buffer.go:50 Update controller.buffer.query-configmap: 1024
2019-06-13 04:14:20.353 +00:00 INFO config/buffer.go:57 Update controller.buffer.query-secret: 1024
2019-06-13 04:14:20.355 +00:00 INFO config/buffer.go:64 Update controller.buffer.pod-event: 1
2019-06-13 04:14:20.357 +00:00 INFO config/buffer.go:71 Update controller.buffer.configmap-event: 1
2019-06-13 04:14:20.361 +00:00 INFO config/buffer.go:78 Update controller.buffer.secret-event: 1
2019-06-13 04:14:20.365 +00:00 INFO config/context.go:28  send module name: cloudhub
2019-06-13 04:14:20.367 +00:00 INFO config/context.go:35 receive module name: controller
2019-06-13 04:14:20.370 +00:00 INFO config/context.go:42 response module name: cloudhub
2019-06-13 04:14:20.372 +00:00 INFO config/kube.go:34 Controller kube master: http://localhost:8080
2019-06-13 04:14:20.374 +00:00 INFO config/kube.go:41 Controller kube config:
2019-06-13 04:14:20.375 +00:00 INFO config/kube.go:46 Controller kube namespace:
2019-06-13 04:14:20.376 +00:00 INFO config/kube.go:51 Controller kube content type: application/vnd.kubernetes.protobuf
2019-06-13 04:14:20.379 +00:00 INFO config/kube.go:56 Controller kube QPS: 5.000000
2019-06-13 04:14:20.382 +00:00 INFO config/kube.go:61 Controller kube burst: 10
2019-06-13 04:14:20.385 +00:00 INFO config/kube.go:66 Controller kube update frequency: 10s
2019-06-13 04:14:20.387 +00:00 INFO config/kube.go:72 Controller kube Node ID:
2019-06-13 04:14:20.388 +00:00 INFO config/kube.go:79 Controller kube Node Name:
2019-06-13 04:14:20.389 +00:00 INFO config/kube.go:86  EdgeSite is false
2019-06-13 04:14:20.391 +00:00 INFO config/load.go:27 update pod status workers: 1
2019-06-13 04:14:20.392 +00:00 INFO config/load.go:34 update node status workers: 1
2019-06-13 04:14:20.393 +00:00 INFO config/load.go:41 query config map workers: 4
2019-06-13 04:14:20.394 +00:00 INFO config/load.go:48 query secret workers: 4
2019-06-13 04:14:20.396 +00:00 INFO config/messagelayer.go:18 message layer: context
2019-06-13 04:14:20.507 +00:00 INFO core/module.go:41 module controller registered
2019-06-13 04:14:20.508 +00:00 INFO config/buffer.go:24 Update devicecontroller.buffer.update-device-status: 1024
2019-06-13 04:14:20.510 +00:00 INFO config/buffer.go:31 Update devicecontroller.buffer.device-event: 1
2019-06-13 04:14:20.511 +00:00 INFO config/buffer.go:38 Update devicecontroller.buffer.device-model-event: 1
2019-06-13 04:14:20.512 +00:00 INFO config/context.go:28 Send module name: cloudhub
2019-06-13 04:14:20.514 +00:00 INFO config/context.go:35 Receive module name: devicecontroller
2019-06-13 04:14:20.516 +00:00 INFO config/context.go:42 Response module name: cloudhub
2019-06-13 04:14:20.517 +00:00 INFO config/kube.go:33 Devicecontroller kube master: http://localhost:8080
2019-06-13 04:14:20.519 +00:00 INFO config/kube.go:40 Devicecontroller kube config:
2019-06-13 04:14:20.520 +00:00 INFO config/kube.go:47 Devicecontroller kube namespace:
2019-06-13 04:14:20.522 +00:00 INFO config/kube.go:54 Devicecontroller kube content type: application/vnd.kubernetes.protobuf
2019-06-13 04:14:20.523 +00:00 INFO config/kube.go:61 Devicecontroller kube QPS: 5.000000
2019-06-13 04:14:20.526 +00:00 INFO config/kube.go:68 Devicecontroller kube burst: 10
2019-06-13 04:14:20.527 +00:00 INFO config/load.go:18 Update device status workers: 1
2019-06-13 04:14:20.531 +00:00 INFO config/messagelayer.go:18 Message layer: context
2019-06-13 04:14:20.533 +00:00 INFO core/module.go:41 module devicecontroller registered
Package: github.com/kubeedge/kubeedge
Version: ca145a1
Revision: ca145a1e9ad437e7b72e7194ce97524c2dd467e9
Goversion: go1.10.2
Branch: version

...


```
because edgecontroller and edge_core use beehive core package: `"github.com/kubeedge/beehive/pkg/core"`
 
that will out put first out put some logs.

so i think , we need change beehive package first. let beehive support version options.

**Comment by rohitsardesai83** [comment]
> @rohitsardesai83 , yes , i know what you mean. maybe this change is not perfect. the best is like kedam:
> this is it`s output:
> 
> ```
> ./keadm/kubeedge version
> Package: github.com/kubeedge/kubeedge
> Version: ca145a1
> Revision: ca145a1e9ad437e7b72e7194ce97524c2dd467e9
> Goversion: go1.10.2
> Branch: version
> ```
> 
> but about edgecontroller and edge_core , now it only output it`s logs:
> 
> this is edgecontroller output:
> 
> ```
> 
> ./cloud/edgecontroller
> 2019/06/13 04:14:20 WARN: empty configurtion from [CommandlineSource]
> 2019/06/13 04:14:20 WARN: empty configurtion from [MemorySource]
> New file source added for configuration: /go/src/github.com/kubeedge/kubeedge/cloud/conf/controller.yamlNew file source added for configuration: /go/src/github.com/kubeedge/kubeedge/cloud/conf/logging.yamlNew file source added for configuration: /go/src/github.com/kubeedge/kubeedge/cloud/conf/modules.yaml/go/src/github.com/kubeedge/kubeedge/cloud/conf/logging.yamllogConfig:&{INFO edgecontroller.log [file stdout] true false  }
> 2019/06/13 04:14:20 RollingPolicy is empty, use default policy[size]
> 2019-06-13 04:14:20.288 +00:00 INFO core/module.go:41 module cloudhub registered
> 2019-06-13 04:14:20.347 +00:00 INFO config/buffer.go:36 Update controller.buffer.update-pod-status: 1024
> 2019-06-13 04:14:20.349 +00:00 INFO config/buffer.go:43 Update controller.buffer.update-node-status: 1024
> 2019-06-13 04:14:20.352 +00:00 INFO config/buffer.go:50 Update controller.buffer.query-configmap: 1024
> 2019-06-13 04:14:20.353 +00:00 INFO config/buffer.go:57 Update controller.buffer.query-secret: 1024
> 2019-06-13 04:14:20.355 +00:00 INFO config/buffer.go:64 Update controller.buffer.pod-event: 1
> 2019-06-13 04:14:20.357 +00:00 INFO config/buffer.go:71 Update controller.buffer.configmap-event: 1
> 2019-06-13 04:14:20.361 +00:00 INFO config/buffer.go:78 Update controller.buffer.secret-event: 1
> 2019-06-13 04:14:20.365 +00:00 INFO config/context.go:28  send module name: cloudhub
> 2019-06-13 04:14:20.367 +00:00 INFO config/context.go:35 receive module name: controller
> 2019-06-13 04:14:20.370 +00:00 INFO config/context.go:42 response module name: cloudhub
> 2019-06-13 04:14:20.372 +00:00 INFO config/kube.go:34 Controller kube master: http://localhost:8080
> 2019-06-13 04:14:20.374 +00:00 INFO config/kube.go:41 Controller kube config:
> 2019-06-13 04:14:20.375 +00:00 INFO config/kube.go:46 Controller kube namespace:
> 2019-06-13 04:14:20.376 +00:00 INFO config/kube.go:51 Controller kube content type: application/vnd.kubernetes.protobuf
> 2019-06-13 04:14:20.379 +00:00 INFO config/kube.go:56 Controller kube QPS: 5.000000
> 2019-06-13 04:14:20.382 +00:00 INFO config/kube.go:61 Controller kube burst: 10
> 2019-06-13 04:14:20.385 +00:00 INFO config/kube.go:66 Controller kube update frequency: 10s
> 2019-06-13 04:14:20.387 +00:00 INFO config/kube.go:72 Controller kube Node ID:
> 2019-06-13 04:14:20.388 +00:00 INFO config/kube.go:79 Controller kube Node Name:
> 2019-06-13 04:14:20.389 +00:00 INFO config/kube.go:86  EdgeSite is false
> 2019-06-13 04:14:20.391 +00:00 INFO config/load.go:27 update pod status workers: 1
> 2019-06-13 04:14:20.392 +00:00 INFO config/load.go:34 update node status workers: 1
> 2019-06-13 04:14:20.393 +00:00 INFO config/load.go:41 query config map workers: 4
> 2019-06-13 04:14:20.394 +00:00 INFO config/load.go:48 query secret workers: 4
> 2019-06-13 04:14:20.396 +00:00 INFO config/messagelayer.go:18 message layer: context
> 2019-06-13 04:14:20.507 +00:00 INFO core/module.go:41 module controller registered
> 2019-06-13 04:14:20.508 +00:00 INFO config/buffer.go:24 Update devicecontroller.buffer.update-device-status: 1024
> 2019-06-13 04:14:20.510 +00:00 INFO config/buffer.go:31 Update devicecontroller.buffer.device-event: 1
> 2019-06-13 04:14:20.511 +00:00 INFO config/buffer.go:38 Update devicecontroller.buffer.device-model-event: 1
> 2019-06-13 04:14:20.512 +00:00 INFO config/context.go:28 Send module name: cloudhub
> 2019-06-13 04:14:20.514 +00:00 INFO config/context.go:35 Receive module name: devicecontroller
> 2019-06-13 04:14:20.516 +00:00 INFO config/context.go:42 Response module name: cloudhub
> 2019-06-13 04:14:20.517 +00:00 INFO config/kube.go:33 Devicecontroller kube master: http://localhost:8080
> 2019-06-13 04:14:20.519 +00:00 INFO config/kube.go:40 Devicecontroller kube config:
> 2019-06-13 04:14:20.520 +00:00 INFO config/kube.go:47 Devicecontroller kube namespace:
> 2019-06-13 04:14:20.522 +00:00 INFO config/kube.go:54 Devicecontroller kube content type: application/vnd.kubernetes.protobuf
> 2019-06-13 04:14:20.523 +00:00 INFO config/kube.go:61 Devicecontroller kube QPS: 5.000000
> 2019-06-13 04:14:20.526 +00:00 INFO config/kube.go:68 Devicecontroller kube burst: 10
> 2019-06-13 04:14:20.527 +00:00 INFO config/load.go:18 Update device status workers: 1
> 2019-06-13 04:14:20.531 +00:00 INFO config/messagelayer.go:18 Message layer: context
> 2019-06-13 04:14:20.533 +00:00 INFO core/module.go:41 module devicecontroller registered
> Package: github.com/kubeedge/kubeedge
> Version: ca145a1
> Revision: ca145a1e9ad437e7b72e7194ce97524c2dd467e9
> Goversion: go1.10.2
> Branch: version
> 
> ...
> ```
> 
> because edgecontroller and edge_core use beehive core package: `"github.com/kubeedge/beehive/pkg/core"`
> 
> that will first output some logs.
> 
> so i think , we need change beehive package first. let beehive support version options.

yes can we have it in a manner like how kubeadm version works : 

```sh
kubeadm version: &version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.2", GitCommit:"17c77c7898218073f14c8d573582e8d2313dc740", GitTreeState:"clean", BuildDate:"2018-10-24T06:51:33Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}

```

**Comment by kadisi** [comment]
> yes can we have it in a manner like how kubeadm version works :

```
kubeadm version: &version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.2", GitCommit:"17c77c7898218073f14c8d573582e8d2313dc740", GitTreeState:"clean", BuildDate:"2018-10-24T06:51:33Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
```

@rohitsardesai83 , ok.

**Comment by kadisi** [comment]
@rohitsardesai83 @sids-b , if we let kubeedge support print version , we need to change beehive. https://github.com/kubeedge/beehive/issues/14.      what do you think?

**Pull request closed without merging**